include/ruby/internal/arithmetic/long.h

```diff
@@ -115,7 +115,7 @@ RB_INT2FIX(long i)
     /* :NOTE: VALUE can be wider than long.  As j being unsigned, 2j+1 is fully
      * defined. Also it can be compiled into a single LEA instruction. */
     const unsigned long j = i;
-    const unsigned long k = 2 * j + RUBY_FIXNUM_FLAG;
+    const unsigned long k = (j << 1) + RUBY_FIXNUM_FLAG;
```

Comment:
(We discussed this a little at work)

It seems like, even though C considers unsigned overflow defined for both `<<` and `*`, UBSAN checks for overflow on `*`, but not `<<`. We figured that this UBSAN behaviour is probably safe to rely on, and avoids having to NO_SANITIZE in this code that is exported in the C extension API. (The output of gcc and clang w/o UBSAN is identical)

👍